src/clib/pio_darray_int.c

```diff
@@ -233,56 +234,60 @@ int write_darray_multi_par(file_desc_t *file, int nvars, int fndims, const int *
-                            break;
-                        case PIO_UINT64:
-                            ierr = nc_put_vara_ulonglong(file->fh, varids[nv], start, count, (unsigned long long*)bufptr);
-                            break;
-                        case PIO_STRING:
-                            ierr = nc_put_vara_string(file->fh, varids[nv], start, count, (const char**)bufptr);
-                            break;
-                        default:
-                            return pio_err(ios, file, PIO_EBADTYPE, __FILE__, __LINE__,
-                                            "Writing variables (number of variables = %d) to file (%s, ncid=%d) using PIO_IOTYPE_NETCDF4P iotype failed. Unsupported variable data type (type=%d)", nvars, pio_get_fname_from_file(file), file->pio_ncid, iodesc->piotype);
-                        }
+                    case PIO_BYTE:
+                        ierr = nc_put_vara_schar(file->fh, varids[nv], start, count, (signed char*)bufptr);
+                        break;
+                    case PIO_CHAR:
+                        ierr = nc_put_vara_text(file->fh, varids[nv], start, count, (char*)bufptr);
+                        break;
+                    case PIO_SHORT:
+                        ierr = nc_put_vara_short(file->fh, varids[nv], start, count, (short*)bufptr);
+                        break;
+                    case PIO_INT:
+                        ierr = nc_put_vara_int(file->fh, varids[nv], start, count, (int*)bufptr);
+                        break;
+                    case PIO_FLOAT:
+                        ierr = nc_put_vara_float(file->fh, varids[nv], start, count, (float*)bufptr);
+                        break;
+                    case PIO_DOUBLE:
+                        ierr = nc_put_vara_double(file->fh, varids[nv], start, count, (double*)bufptr);
+                        break;
+                    case PIO_UBYTE:
+                        ierr = nc_put_vara_uchar(file->fh, varids[nv], start, count, (unsigned char*)bufptr);
+                        break;
+                    case PIO_USHORT:
+                        ierr = nc_put_vara_ushort(file->fh, varids[nv], start, count, (unsigned short*)bufptr);
+                        break;
+                    case PIO_UINT:
+                        ierr = nc_put_vara_uint(file->fh, varids[nv], start, count, (unsigned int*)bufptr);
+                        break;
+                    case PIO_INT64:
+                        ierr = nc_put_vara_longlong(file->fh, varids[nv], start, count, (long long*)bufptr);
+                        break;
+                    case PIO_UINT64:
+                        ierr = nc_put_vara_ulonglong(file->fh, varids[nv], start, count, (unsigned long long*)bufptr);
+                        break;
+                    case PIO_STRING:
+                        ierr = nc_put_vara_string(file->fh, varids[nv], start, count, (const char**)bufptr);
+                        break;
+                    default:
```

Comment:
It would be better to put a break for the default case.

From The C programming language - Second edition (K&R 2):
Chapter 3.4 Switch
> As a matter of good form, put a break after the last case (the default here) even though it's logically unnecessary. Some day when another case gets added at the end, this bit of defensive programming will save you.

Reply:
ok I will add the break, but adding a case after default is also poor form.

Reply:
This should be fixed now